src/common/snippets/src/op/subgraph.cpp

```diff
@@ -730,7 +717,23 @@ snippets::Schedule Subgraph::generate(const std::vector<pass::Manager::Positione
     const auto& lowering_result = m_generator->generate(linear_ir, linear_ir.get_config(), compile_params);
     const auto ptr = lowering_result.binary_code;
 
-    return {master_shape, false /*canBeLinearized*/, ptr};
+
+    VectorDims work_domain{1};
+    for (const auto& expr : linear_ir.get_IO_ops()) {
+        if (expr->get_type() == snippets::lowered::IOExpression::io_type::OUTPUT) {
+            const auto& shape = utils::get_planar_vdims(expr->get_input_port_descriptor(0));
+            OPENVINO_ASSERT(std::none_of(shape.begin(), shape.end(),
+                                         [](size_t d) {return d == snippets::IShapeInferSnippets::DYNAMIC_DIMENSION; }),
+                            "Failed to calculate work_domain for dynamic shapes");
+            OPENVINO_ASSERT(ov::snippets::broadcast_merge_into(work_domain, shape),
+                            "Failed to merge input shapes into work_domain");
+        }
+    }
```

Comment:
Corporate wants you to find the difference between these two pictures ©️ 😁 
You're right, thanks! This can be replaced with `linear_ir->get_master_shape()`. That's one of the reasons why I wrapped this logic as a separate method